src/import-data/ImportData.test.ts

```diff
@@ -0,0 +1,448 @@
+
+type WorkspaceUtilsExports = typeof import('src/components/workspace-utils');
+jest.mock('src/components/workspace-utils', (): WorkspaceUtilsExports => {
+  return {
+    ...jest.requireActual<WorkspaceUtilsExports>('src/components/workspace-utils'),
+    useWorkspaces: jest.fn(),
+  };
+});
+
+type AjaxExports = typeof import('src/libs/ajax');
+type AjaxContract = ReturnType<AjaxExports['Ajax']>;
+jest.mock('src/libs/ajax');
+
+type NavExports = typeof import('src/libs/nav');
+jest.mock('src/libs/nav', (): NavExports => {
+  return {
+    ...jest.requireActual<NavExports>('src/libs/nav'),
+    goToPath: jest.fn(),
+    useRoute: jest.fn(),
+  };
+});
+
+type NotificationsExports = typeof import('src/libs/notifications');
+jest.mock('src/libs/notifications', (): NotificationsExports => {
+  return {
+    ...jest.requireActual<NotificationsExports>('src/libs/notifications'),
+    notify: jest.fn(),
+  };
+});
+
+type DataBrowserUtilsExports = typeof import('src/pages/library/dataBrowser-utils');
+jest.mock('src/pages/library/dataBrowser-utils', (): DataBrowserUtilsExports => {
+  return {
+    ...jest.requireActual<DataBrowserUtilsExports>('src/pages/library/dataBrowser-utils'),
+    useDataCatalog: jest.fn(),
+  };
+});
+
+interface SetupOptions {
+  mockAjax: DeepPartial<AjaxContract>;
+  queryParams: { [key: string]: unknown };
+}
+
+const setup = (opts: SetupOptions) => {
+  const { mockAjax, queryParams } = opts;
+
+  const mockAjaxWithCommonFunctions: DeepPartial<AjaxContract> = {
+    ...mockAjax,
+    Billing: {
+      listProjects: jest.fn().mockResolvedValue([{}]),
+      ...mockAjax.Billing,
+    },
+    FirecloudBucket: {
+      getTemplateWorkspaces: jest.fn().mockResolvedValue([]),
+      ...mockAjax.FirecloudBucket,
+    },
+    Metrics: {
+      captureEvent: jest.fn(),
+      ...mockAjax.Metrics,
+    },
+  };
+  asMockedFn(Ajax).mockImplementation(() => mockAjaxWithCommonFunctions as AjaxContract);
+
+  asMockedFn(useRoute).mockReturnValue({
+    query: queryParams,
+  });
+
+  render(h(ImportData));
+};
+
+const importIntoExistingWorkspace = async (user: UserEvent, workspaceName: string): Promise<void> => {
+  const existingWorkspace = screen.getByText('Start with an existing workspace', { exact: false });
+  await user.click(existingWorkspace);
+
+  const workspaceSelect = new SelectHelper(screen.getByLabelText('Select a workspace'), user);
+  await workspaceSelect.selectOption(workspaceName);
+
+  await user.click(screen.getByRole('button', { name: 'Import' }));
+};
+
+describe('ImportData', () => {
+  beforeEach(() => {
+    // Arrange
+    asMockedFn(useWorkspaces).mockReturnValue({
+      workspaces: [defaultAzureWorkspace, defaultGoogleWorkspace],
+      loading: false,
+      refresh: () => Promise.resolve(),
+    });
+
+    asMockedFn(useDataCatalog).mockReturnValue({
+      dataCatalog: [],
+      loading: false,
+      refresh: () => Promise.resolve(),
+    });
+  });
+
+  describe('PFB imports', () => {
```

Comment:
Can you drop the outer `describe` here and outdent?  I think the `it` clause does a good job covering the descriptiveness.

Reply:
Yes, but I also kind of like having the test organized by import type.

Reply:
Did some regrouping - organized PFB, BagIt, and Rawls JSON all under a `describe('files', ...`.